reminder/database.py

```diff
@@ -0,0 +1,196 @@
+        """
+        origin_date = datetime.now()
+        if remind_date < origin_date:
+            raise ValueError
+
+        item = ReminderItem(
+            guild_id=guild_id,
+            author_id=author_id,
+            remind_id=remind_id,
+            permalink=permalink,
+            message=message,
+            origin_date=origin_date,
+            remind_date=remind_date,
+            status=ReminderStatus.WAITING,
+        )
+
+        session.add(item)
+        session.commit()
+        return item
+
+    @staticmethod
+    def get_all(
+        guild_id: int = None,
+        idx: int = None,
+        remind_id: int = None,
+        status: ReminderStatus = None,
+        min_origin_date: datetime = None,
+        max_origin_date: datetime = None,
+        min_remind_date: datetime = None,
+        max_remind_date: datetime = None,
+    ) -> List[ReminderItem]:
+        """Retreives List of ReminderItem filtered by Guild ID.
+
+        Args:
+            guild_id: Guild whose items are to be returned.
+            idx: ID of reminder item
+            remind_id: ID of reminded user whose items are to be returned.
+            status: Status of items to be returned
+            min_origin_date: Filter items created after this date.
+            max_origin_date: Filter items created before this date.
+            min_remind_date: Filter items being reminded after this date.
+            max_remind_date: Filter items being reminded before this date.
+
+        Returns:
+            :class:`List[ReminderItem]`: The retrieved reminder items ordered by remind_date descending.
+        """
+        query = session.query(ReminderItem)
+
+        if guild_id is not None:
+            query = query.filter_by(guild_id=guild_id)
+
+        if idx is not None:
+            query = query.filter_by(idx=idx)
+
+        if remind_id is not None:
+            query = query.filter_by(remind_id=remind_id)
+
+        if status is not None:
+            query = query.filter_by(status=status)
+
+        if min_origin_date is not None:
+            query = query.filter(ReminderItem.origin_date > min_origin_date)
+
+        if max_origin_date is not None:
+            query = query.filter(ReminderItem.origin_date < max_origin_date)
+
+        if min_remind_date is not None:
+            query = query.filter(ReminderItem.remind_date > min_remind_date)
+
+        if max_remind_date is not None:
+            query = query.filter(ReminderItem.remind_date < max_remind_date)
+
+        query = query.order_by(ReminderItem.remind_date.desc())
+
+        return query.all()
+
+    def delete(self):
+        """
+        Deletes the item from the database.
+        """
+        session.delete(self)
+        session.commit()
+
+    def reschedule(self, new_date: datetime):
+        """
+        Change reminding date and time
+
+        Args:
+            new_date: New date reminder should be sent
+        """
+        self.remind_date = new_date
```

Comment:
These methods that just modify a single item attribute aren't necessary I believe. 
I don't use them in my modules anymore. Instead I set the attribute in the code and call the `save()` method